Userland/Libraries/LibJS/Runtime/VM.cpp

```diff
@@ -207,18 +207,15 @@ void VM::assign(const NonnullRefPtr<BindingPattern>& target, Value value, Global
         if (!iterator)
             return;
 
-        size_t index = 0;
-        while (true) {
+        for (size_t i = 0; i < binding.properties.size(); i++) {
             if (exception())
                 return;
 
-            if (index >= binding.properties.size())
-                break;
-
-            auto pattern_property = binding.properties[index];
-            ++index;
+            auto pattern_property = binding.properties[i];
 
             if (pattern_property.is_rest) {
+                VERIFY(i == binding.properties.size());
```

Comment:
This check seems impossible, you're verifying that the element you just grabbed is one past the end?

Reply:
It's left over from a previous iteration of the code. This is a draft, after all -- I wasn't expecting a review so soon :)

Reply:
Me: sees an interesting draft
Me: ignores "draft"